core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -885,7 +901,7 @@ public void approveAll(Set<Statement> approve, Set<Resource> approveContexts) {
 				approved = createEmptyModel();
 			}
 			approved.addAll(approve);
-			approvedEmpty = approved == null || approved.isEmpty();
+			approvedEmpty = false;
```

Comment:
Maybe `approvedEmpty` should be set to `approve.isEmpty()` since it's technically allowed to pass in an empty set. 

Same for `deprecatedEmpty`.

Reply:
`approve.isEmpty()` was the problem here because it leads to flushing and committing the underlying transaction of MemoryOverflowModel. Can `approved` be empty at this point?

Reply:
It used to be `approvedEmpty = approved == null || approved.isEmpty();`. Couldn't it instead be `approvedEmpty = approve.isEmpty();`? Notice the difference that one uses approve**d** and the other just approve.